### ISSUE TITLE
Error updating SealedSecret namespace/secret status: sealedsecrets.bitnami.com "secret" not found

### ISSUE THREAD

**Issue by psdonas** [bitnami-labs/sealed-secrets#1021]
**Which component**:
The name (and version) of the affected component (controller or kubeseal)
helm.sh/chart=sealed-secrets-2.1.6
kubeseal version: v0.18.0

**Describe the bug**
This must be a bug, because all the secrets are working fine. When I look into the pod logs I receive this error for all secrets managed by sealedsecrets: 
`Error updating SealedSecret namespace/secret status: sealedsecrets.bitnami.com "secret" not found`
I also should mention that adding status flag did not resolve the issue. 

**To Reproduce**

2. Run the command '....'
3. Wait for '....'

**Expected behavior**

**Version of Kubernetes**:

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.14-gke.2700", GitCommit:"2211759978749a9c2935fe2f8db1e6823c3c0bb1", GitTreeState:"clean", BuildDate:"2022-07-26T09:20:06Z", GoVersion:"go1.16.15b7", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

**Comment by alemorcuq** [comment]
Could you try using the most recent version of the chart? Recently we fixed some issues with the `status` subresource.

**Comment by Mahagon** [comment]
I think i have the same issue:

```
2022/10/25 07:01:11 Updating monitoring/alertmanager-auth
2022/10/25 07:01:12 Event(v1.ObjectReference{Kind:"SealedSecret", Namespace:"monitoring", Name:"alertmanager-auth", UID:"40ab9f49-713c-43f4-95dc-14bb51c010ca", APIVersion:"bitnami.com/v1alpha1", ResourceVersion:"282073243", FieldPat
2022/10/25 07:01:12 Error updating SealedSecret monitoring/alertmanager-auth status: sealedsecrets.bitnami.com "alertmanager-auth" not found 
```

Kubernetes Version: v1.23.12
Kubeseal Version: v0.18.5
Chart Version: 2.7.0
Image: docker.io/bitnami/sealed-secrets-controller:v0.19.1

Secrets are working fine, even if i deploy new ones

**Comment by Mahagon** [comment]
It looks like #964 got merged on the 20th, so its not in the 2.7.0 chart, which was release on the 17th.

**Comment by tewfik-ghariani** [comment]
@Mahagon That commit actually made it to the 2.7.0 chart because it was merged on the 20th of September while the latest release was published on the 17th October ^^

**Comment by Mahagon** [comment]
@tewfik-ghariani oha, i should learn reading...


Hm strange, my crd didn't update after applying chart sealed-secrets/sealed-secrets:2.7.0 

<details>
  <summary>kubectl diff crd</summary>

```shell
kubectl diff --force-conflicts --server-side -f https://raw.githubusercontent.com/bitnami-labs/sealed-secrets/v0.19.1/helm/sealed-secrets/crds/bitnami.com_sealedsecrets.yaml
diff -u -N /tmp/LIVE-2473281265/apiextensions.k8s.io.v1.CustomResourceDefinition..sealedsecrets.bitnami.com /tmp/MERGED-3165320711/apiextensions.k8s.io.v1.CustomResourceDefinition..sealedsecrets.bitnami.com
--- /tmp/LIVE-2473281265/apiextensions.k8s.io.v1.CustomResourceDefinition..sealedsecrets.bitnami.com    2022-10-25 15:37:02.726937331 +0200
+++ /tmp/MERGED-3165320711/apiextensions.k8s.io.v1.CustomResourceDefinition..sealedsecrets.bitnami.com  2022-10-25 15:37:02.730270695 +0200
@@ -4,7 +4,7 @@
   annotations:
     controller-gen.kubebuilder.io/version: v0.9.2
   creationTimestamp: "2022-09-14T09:31:35Z"
-  generation: 1
+  generation: 2
   name: sealedsecrets.bitnami.com
   resourceVersion: "224991708"
   uid: 4f349f85-2bdd-4966-b88a-b17a2ca50487
@@ -63,6 +63,7 @@
                     type: object
                   metadata:
                     description: 'Standard object''s metadata. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata'
+                    nullable: true
                     type: object
                     x-kubernetes-preserve-unknown-fields: true
                   type:
@@ -124,6 +125,8 @@
         type: object
     served: true
     storage: true
+    subresources:
+      status: {}
 status:
   acceptedNames:
     kind: SealedSecret
```

</details>

**Comment by Mahagon** [comment]
created a procedure to upgrade the crd first before deploying the helm chart

https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#install-a-crd-declaration-before-using-the-resource

Error msg is gone :)
